app/controllers/comments_controller.rb

```diff
@@ -142,11 +152,37 @@ def check_permission_to_delete
     access_denied(redirect: @comment) unless logged_in_as_admin? || current_user_owns?(@comment) || current_user_owns?(@comment.ultimate_parent)
   end
 
-  # Comments cannot be edited after they've been replied to
+  # Comments cannot be edited after they've been replied to or if they are frozen.
   def check_permission_to_edit
-    return if @comment&.count_all_comments&.zero?
-    flash[:error] = ts("Comments with replies cannot be edited")
-    redirect_to(request.env["HTTP_REFERER"] || root_path)
+    if @comment&.iced?
+      flash[:error] = ts("Frozen comments cannot be edited.")
+      redirect_to(request.env["HTTP_REFERER"] || root_path)
+    elsif !@comment&.count_all_comments&.zero?
+      flash[:error] = ts("Comments with replies cannot be edited")
+      redirect_to(request.env["HTTP_REFERER"] || root_path)
+    end
+  end
+
+  # Comments on works can be frozen by admins with proper authorization or the
+  # work creator.
+  # Comments on tags can be frozen by admins with proper authorization.
+  # Comments on admin posts can be frozen by any admin.
+  def check_permission_to_freeze
+    return if permission_to_modify_frozen_status
+
+    flash[:error] = ts("Sorry, you don't have permission to freeze that comment thread.")
+    redirect_to(request.env["HTTP_REFERER"] || root_path) && return
```

Comment:
You don't need `&& return` here, since it's the end of the function. (Ditto with all of the places it's used below, as well.)